compiler/rustc_target/src/spec/base/apple/mod.rs

```diff
@@ -105,6 +105,7 @@ pub(crate) fn base(
 ) -> (TargetOptions, StaticCow<str>, StaticCow<str>) {
     let opts = TargetOptions {
         abi: abi.target_abi().into(),
+        llvm_floatabi: Some(FloatAbi::Hard),
```

Comment:
I assume all the Apple targets use hardfloats? This is about armv7s_apple_ios.rs and armv7k_apple_watchos.rs. LLVM switches to hardfloats for `(TargetTriple.isOSBinFormatMachO() && TargetTriple.getSubArch() == Triple::ARMSubArch_v7em)`, not sure if that applies here.

Cc @deg4uss3r @madsmtm

Reply:
That is my understanding, yeah, and matches what Clang reports when you try to use a different float ABI:

```console
$ touch foo.c
$ xcrun -sdk watchos clang -target armv7k-apple-watchos -S -c foo.c -mfloat-abi=hard # Succeeds
$ xcrun -sdk watchos clang -target armv7k-apple-watchos -S -c foo.c -mfloat-abi=soft # Fails
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7k-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7k-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7k-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7k-apple-watchos'
$ xcrun -sdk watchos clang -target armv7s-apple-watchos -S -c foo.c -mfloat-abi=hard # Succeeds
$ xcrun -sdk watchos clang -target armv7s-apple-watchos -S -c foo.c -mfloat-abi=soft # Fails
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7s-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7s-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7s-apple-watchos'
clang: error: unsupported option '-mfloat-abi=soft' for target 'thumbv7s-apple-watchos'
```